### DISCUSSION TITLE
"Remove extra config" button not enabled when dealing with jobs

### DISCUSSION THREAD

**Discussion by ghost** [dagster-io/dagster#19645] [Q&A]
Why is the "Remove extra config" button not enabled when dealing with jobs, but is enabled when dealing with graphs?
Maybe I'm doing something wrong (probably).

**Not using a graph:**
```
class OpsConfig(Config):
    env: str = None

test_env = RunConfig(
    ops={
        "pipeline_a": OpsConfig(env="test"),
        "pipeline_b": OpsConfig(env="test")
    }
)

@op(ins={"start": In(Nothing)})
def pipeline_a(config: OpsConfig):
    print(f"pipeline_a in {config.env}")

@op(ins={"start": In(Nothing)})
def pipeline_b(config: OpsConfig):
    print(f"pipeline_b in {config.env}")

@job(name="Testing_Scenario", config=test_env)
def testing_scenario():
    a = pipeline_a()
    b = pipeline_b()
```

**Here is the UI:**
![image](https://github.com/dagster-io/dagster/assets/135677220/3b68f850-8c98-48c5-b5f9-34717bd97fbf)

**Using a graph:**
```
class OpsConfig(Config):
    env: str = None

test_env_graph = RunConfig(
    ops={
        "test_graph": {
            "ops": {
                "pipeline_a": OpsConfig(env="test"),
                "pipeline_b": OpsConfig(env="test")
            }
        }
    }
)

@graph
def test_graph():
    a = pipeline_a()
    b = pipeline_b()


@job(name="Testing_Scenario_Graph", config=test_env_graph)
def testing_scenario_graph():
    test_graph()

```

**UI:**
![image](https://github.com/dagster-io/dagster/assets/135677220/4c972c90-f045-4e42-b642-f29ba2a1cdce)

**I can edit the launchpad and manually remove the extra configuration, but I thought it would work similarly when using `@graph.`**

**Comment by garethbrickman** [comment]
hey gemini-trek, I noticed you asked about this in the[ #ask-ai Slack channel](https://dagster.slack.com/archives/C066HKS7EG1/p1707255622466569?thread_ts=1707255383.046999&cid=C066HKS7EG1). Did the responses provided there help to answer/resolve your question?

**Reply by ghost** [reply]
Hi @garethbrickman, thank you for your reply. The explanation was interesting, but I think I need more experience regarding how dagster interprets job selection. I'll ask more questions on the channel.

So what I understood is that there is actually no error in my configuration or in the way dagster works in this case. It's just a conceptual issue(?).

I mean, logically for me it would work the same way in both scenarios, since dagster recognizes the initial configuration, but I didn't want a graph within a job since it adds an extra step when selecting jobs.
